dao.py

```diff
@@ -292,6 +292,29 @@ def get_tournament_by_id(self, id):
         '''id must be an ObjectId'''
         return M.Tournament.load(self.tournaments_col.find_one({'_id': id}), context='db')
 
+    def get_tournament_by_id_and_match_id(self, tournament_id, match_id):
+        tourney_m = M.Tournament.load(self.tournaments_col.find_one({'_id': tournament_id}, {'matches', 1}), context='db')
+        for match in tourney_m.matches:
+            if match_id == match.id:
+                return match
+
+    def set_match_exclusion_by_tournament_id_and_match_id(self, tournament_id, match_id, excluded):
+        # TODO ENHANCE THIS ALGORITHM TO ONLY UPDATE MATCH
+        match_updated = False
+        new_matches = []
+        tourney_m = \
+            M.Tournament.load(self.tournaments_col.find_one({'_id': tournament_id}), context='db')
+
+        for match in tourney_m.matches:
+            if int(match_id) == int(match.match_id):
```

Comment:
I will try-catch here, however this should never happen. Match id's are set by the automation, so unless an admin changes it directly to something else, we will never error here.